src/features/Chat/chatSlice.ts

```diff
@@ -204,11 +204,76 @@ const getNextParams = (
 
 const selectChatState = ({ chats }: RootState) => chats;
 
+const compareMessagesByTimeCreated = (
+  messageA: AppMessage,
+  messageB: AppMessage,
+): number =>
+  new Date(messageB.created).getTime() - new Date(messageA.created).getTime();
+
+const sortMessagesByDateDescending = (messages: AppMessage[]): AppMessage[] =>
+  [...messages].sort(compareMessagesByTimeCreated);
+
+const sortChats = (a: ChatBuddy, b: ChatBuddy): number => {
+  const aHasNoMessages = a.messages.length === 0;
+  const bHasNoMessages = b.messages.length === 0;
+
+  // If both chats have no messages, they are considered equal
+  if (aHasNoMessages && bHasNoMessages) return 0;
+
+  // If only a has no messages, it should come first
+  if (aHasNoMessages) return -1;
+
+  // If only b has no messages, it should come first
+  if (bHasNoMessages) return 1;
+
+  const mostRecentA = sortMessagesByDateDescending(a.messages)[0];
+  const mostRecentB = sortMessagesByDateDescending(b.messages)[0];
+  return compareMessagesByTimeCreated(mostRecentA, mostRecentB);
+};
+
+export const selectChats = createSelector(
+  selectChatState,
+  ({ activeFolder, chats }) =>
+    Object.values(chats)
+      .filter(chat => chat.status === activeFolder)
+      .sort(sortChats),
+);
+
+export const selectAnyChats = createSelector(
+  selectChatState,
+  ({ chats }) => Object.values(chats).length > 0,
+);
+
 export const selectActiveChat = createSelector(
   selectChatState,
   ({ activeChatId, chats }) => (activeChatId ? chats[activeChatId] : null),
 );
 
+export const selectIsActiveChat = createSelector(
+  selectChatState,
+  ({ activeChatId, chats }) => Boolean(activeChatId && chats[activeChatId]),
+);
+
+// Returns most recent unread chat, or the most recent if all are read
+export const selectDefaultChat = createSelector(
+  selectChatState,
+  ({ activeFolder, chats }) => {
+    const sortedChats = Object.values(chats)
+      .filter(chat => chat.status === activeFolder)
+      .sort(sortChats);
+
+    const unreadChats = sortedChats.filter(chat => {
+      if (chat.status !== 'ok') return false;
+      for (const message of chat.messages) {
+        if (!message.opened) return true;
+      }
```

Comment:
I would want to refactor this so we would not use `for...of`

Maybe instead `return chat.messages.some(message => !message.opened)`

Reply:
Good suggestion! Fixed!